examples/scenes/src/cosmic_text_scene.rs

```diff
@@ -0,0 +1,452 @@
+// Copyright 2024 the Vello Authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! An example integrating Cosmic text with Vello.
+//! This renders a read-only snapshot of a simplified text editor using system fonts.
+
+use crate::{SceneParams, TestScene};
+
+use vello::kurbo::{Affine, Point, Rect, Size};
+use vello::peniko::{Blob, Color, Fill, Font};
+use vello::{Glyph, Scene};
+
+use cosmic_text::fontdb::ID;
+use cosmic_text::{
+    Attrs, Buffer, Cursor, Edit, Editor, FontSystem, LayoutRun, Metrics, Selection, Shaping,
+};
+
+use unicode_segmentation::UnicodeSegmentation;
+
+use std::cmp;
+use std::collections::HashMap;
+use std::sync::Arc;
+
+pub struct CosmicData {}
+
+impl TestScene for CosmicTextSceneState {
+    fn render(&mut self, scene: &mut Scene, _scene_params: &mut SceneParams) {
+        let buffer_glyphs = &self.buffer_glyphs;
+        let vello_fonts = &self.vello_fonts;
+
+        let text_transform = Affine::translate((500.0, 300.0));
+
+        // Draw the Glyphs
+        for buffer_line in &buffer_glyphs.buffer_lines {
+            for glyph_highlight in &buffer_line.glyph_highlights {
+                scene.fill(
+                    Fill::NonZero,
+                    text_transform,
+                    buffer_glyphs.glyph_highlight_color,
+                    None,
+                    glyph_highlight,
+                );
+            }
+
+            if let Some(cursor) = &buffer_line.cursor {
+                scene.fill(
+                    Fill::NonZero,
+                    text_transform,
+                    buffer_glyphs.cursor_color,
+                    None,
+                    cursor,
+                );
+            }
+
+            for glyph_run in &buffer_line.glyph_runs {
+                let font = vello_fonts.get(&glyph_run.font).unwrap();
+                let glyph_color = glyph_run.glyph_color;
+                let glyphs = glyph_run.glyphs.clone();
+                scene
+                    .draw_glyphs(font)
+                    .font_size(buffer_glyphs.font_size)
+                    .brush(glyph_color)
+                    .transform(text_transform)
+                    .draw(Fill::NonZero, glyphs.into_iter());
+            }
+        }
+    }
+}
+
+pub struct CosmicTextSceneState {
```

Comment:
"State" is rarely a meaningful suffix; I think this would be clearer if it were removed.